src/shell/boards/board.tsx

```diff
@@ -37,45 +35,19 @@ const BoardContainer = styled.div<{ show: boolean }>`
 
 export const AppBoard = ({ board }: { board: Board }): React.JSX.Element => {
 	const current = useBoardStore((s) => s.current);
-	// eslint-disable-next-line react-hooks/exhaustive-deps
-	const history = useMemo(() => createMemoryHistory({ initialEntries: [board.url] }), []);
 	const boardViews = useAppStore((s) => s.views.board);
-	const windowHistory = useHistory();
-	const route = useMemo(() => {
-		const view = find(boardViews, (v) => v.id === board.url || startsWith(board.url, v.route));
+	const boardView = useMemo(() => {
+		const view = find(boardViews, (v) => v.id === board.boardViewId);
 		if (view)
 			return (
-				<Route key={view.id} path={view.route}>
-					<AppContextProvider key={view.id} pkg={view.app}>
-						<BoardProvider id={board.id}>
-							<view.component windowHistory={windowHistory} />
-						</BoardProvider>
-					</AppContextProvider>
-				</Route>
+				<AppContextProvider key={view.id} pkg={view.app}>
+					<BoardProvider id={board.id}>
+						<view.component />
+					</BoardProvider>
+				</AppContextProvider>
 			);
```

Comment:
We could extract this component and just memoize the return value of the find. Then in the component return block we could conditionally render the AppContextProvider and the rest of it based on the memoized view. It seems cleaner to me.